blas/testblas/common.go

```diff
@@ -260,7 +260,16 @@ func zsame(x, y []complex128) bool {
 	}
 	for i, v := range x {
 		w := y[i]
-		if v != w && !math.IsNaN(real(v)) && !math.IsNaN(imag(v)) && !math.IsNaN(real(w)) && !math.IsNaN(imag(w)) {
+		if math.IsNaN(real(v)) && math.IsNaN(imag(v)) && math.IsNaN(real(w)) && math.IsNaN(imag(w)) {
```

Comment:
Could we not have something like 

```
func sameFloat64(a, b float64) bool {
    return a == b || (math.IsNaN(a) && math.IsNaN(b))
}
```

and then

```
for i, v := range x {
    w := y[i]
    if !sameFloat64(real(v), real(w)) || !sameFloat64(imag(v), imag(w)) {
        return false
    }
}
```

Reply:
I added `sameFloat64` and then `sameComplex128` which uses it.